spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/SparkZOrderStrategy.java

```diff
@@ -133,9 +133,23 @@ public SparkZOrderStrategy(Table table, SparkSession spark, List<String> zOrderC
           "Cannot perform ZOrdering, all columns provided were identity partition columns and cannot be used.");
     }
 
+    validateColumnsExistence(table, spark, zOrderColNames);
+
     this.zOrderColNames = zOrderColNames;
   }
 
+  private void validateColumnsExistence(Table table, SparkSession spark, List<String> colNames) {
+    boolean caseSensitive = Boolean.parseBoolean(spark.conf().get("spark.sql.caseSensitive"));
```

Comment:
Rather than this we can use the session resolver class which automatically cases or not based on the conf

Reply:
Can you point me to an example code? I am not aware of it and couldn't find it either. 

Also, I did like this because some part of the iceberg code is doing similarly. (by lookingup `spark.sql.caseSensitive` in the code) . Maybe I could replace them too.

Reply:
Other places in Iceberg do this because it is Iceberg binding expressions. Where we can use the Spark resolver, we should.

Reply:
https://github.com/apache/iceberg/blob/master/spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AlignRowLevelCommandAssignments.scala

Has an example of how to use this, see `conf.resolver`

Reply:
I got `Function2<String, String, Object> resolver = SQLConf.get().resolver();` in java but still couldn't figure out how to  use this in my case. 

Scala I can do find and match. But not sure in java.

Reply:
To compare two strings A and B
```java
(Boolean) SQLConf.get().resolver().apply(stringA, stringB)

  @Test
  public void useResolver() {
    Assert.assertTrue((Boolean) SQLConf.get().resolver().apply("Hello","hello"));
    Assert.assertTrue((Boolean) SQLConf.get().resolver().apply("Hello","HeLLO"));
    Assert.assertFalse((Boolean) SQLConf.get().resolver().apply("yellow","HeLLO"));
  }

```

Reply:
But here the problem statement is to find the column using the `currentCase` or `lowerCase` of a column name from a map having column name as keys. So, I think resolver (equals()) will not help here.

Reply:
After seeing what this is doing with the `caseSensitive` value, I think this is reasonable. This is using the value to decide whether to use `findField` or `caseInsensitiveFindField`. I don't see a good way to pass a resolver for those cases, so this seems fine to me.